Automattic/wp-calypso
# Launchpad: Add domain upsell task for Newsletter

### Estimated Time to Test / Review:
- Test -> Short
- Review -> Short

### Proposed Changes
This PR is a follow up to https://github.com/Automattic/wp-calypso/pull/73440 adding the domain upsell task to the Newsletter flow

### Testing
* Checkout this branch
* Create a new `newsletter` flow site, select a free domain and free plan.
* Navigate to the launchpad and confirm that the `Choose a domain` task is visible in the task checklist.
* Click the task and confirm that it navigates to `domains/add/{siteSlug}?domainAndPlanPackage=true`
* Select a new domain and paid plan.
* Navigate back to the launchpad and confirm the `Choose a domain` task is no longer visible
* Confirm the launchpad test cases are still passing: `npm run test-client client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/test`
